File: api/src/main/java/fr/huiitre/tools/modules/core/auth/api/dto/UserModuleDto.java
package fr.huiitre.tools.modules.core.auth.api.dto;

public class UserModuleDto {

    private final String code;
    private final String name;
    private final String description;
    private final String role;

    public UserModuleDto(String code, String name, String description, String role) {
        this.code = code;
        this.name = name;
        this.description = description;
        this.role = role;
    }

    public String getCode() { return code; }
    public String getName() { return name; }
    public String getDescription() { return description; }
    public String getRole() { return role; }
}

File: api/Modules/Users/Application/Dto/UserModuleDto.cs
namespace Tools.Api.Modules.Users.Application.Dto;

// Reflète tools_core.module, complété par le rôle que l'utilisateur détient sur ce module.
// La clé primaire de user_module_role est (user_id, module_id) : un utilisateur n'y détient
// qu'un rôle, d'où une valeur unique et non une liste.
public sealed record UserModuleDto(
    long Id,
    string Code,
    string Name,
    string? Description,
    bool Active,
    RoleDto Role
);
